site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -395,7 +401,7 @@ public function ajaxEditCourseMaterialsFiles(bool $flush = true): JsonResponse {
                 }
 
                 foreach ($course_material->getSections() as $section) {
-                    if (!in_array($section->getSectionId(), $keep_ids)) {
+                    if (!in_array($section->getSectionId(), $keep_ids) && !in_array($section->getSectionId(), $partial_sections)) {
```

Comment:
Changes look good. One thing I didnt understand is why you need the partialSections here. I thought partial sections is solely a front end thing and it wont get inserted into the database. If that is the case do you need the check here. I might be missing something.

Reply:
This takes care of the destructive nature of folder edits that was present before.
Assume a folder X having materials A and B.
A: Section 1, Section 2
B: Section 2
Now, if you were to edit folder X and set section 2, then the earlier behavior was that Section 1 would get unset for material A. However, now this will not be the case as sections that are gray colored don't get unset.
But what if you want to have destructive behavior (unset Section 1 and only keep Section 2) ?:
Just click on that section twice, first time it will turn from gray to blue and second time from blue to white (unchecked), now it is not a partial section and will be removed from the materials present inside the folder.

Reply:
Ah makes sense!.